src/api/app/models/repository.rb

```diff
@@ -123,6 +123,28 @@ def expand_all_repositories
     repositories.uniq
   end
 
+  def cycles(arch)
+    # skip all packages via package=- to speed up the api call, we only parse the cycles anyway
+    deps = BuilddepInfo.find(project: project.name, package: '-', repository: name, arch: arch)
+    cycles = deps.each(:cycle).map { |cycle| cycle.each(:package).map(&:text) }
+
+    merged_cycles = []
+    cycles.each do |cycle|
+      # We look up all other cycles that intersect with this cycle
+      intersecting_cycles = merged_cycles.select { |another_cycle| (cycle & another_cycle).any? }
+      intersecting_cycles.each do |intersecting_cycle|
+        # We remove the intersecting cycle from the merged cycles
+        deleted = merged_cycles.delete(intersecting_cycle)
+        # We merge the intersecting cycle with this cycle
```

Comment:
Is this comment really needed? I think it is pretty clear what the code does....